src/System/extract_system_rep.rs

```diff
@@ -36,6 +38,23 @@ pub fn create_executable_query<'a>(
                 sys1: left.compile(dim)?,
                 sys2: right.compile(dim)?,
             }))},
+            QueryExpression::Reachability(automata, start, end) => {
+                
+                let mut quotient_index = None;
+                let machine = get_system_recipe(automata, component_loader, &mut dim, &mut quotient_index);
+                validate_reachability(&machine, &start, &end);
```

Comment:
clippy complaints, "this expression creates a reference which is immediately dereferenced by the compiler"